### PR TITLE
New Feature - Joint Public Comment Period

### PR DESCRIPTION
**Pull request by asanchezr**
Includes:

* EPIC-960 Joint PCP - Can Download Comments
* EPIC-961 Joint PCP - CEAA: Can download Comments
* EPIC-979 Joint PCP - Mandatory name and location
* EPIC-931 Joint PCP - Feature Implementation: Submit a Comment
* EPIC-932 Joint PCP - Modify PCP Creation Page to Joint OR Regular
* EPIC-935 Joint PCP - Hook up Uploader
* EPIC-937 Joint PCP - Public View
* EPIC-976 PCP (and Joint PCP) - Banner Modifications
* EPIC-1015 Joint PCP - comment details dialog does not show attachments for public users
* EPIC-1017 Joint PCP - Attachment icon not showing on PCP comment display page
* EPIC-1021 Joint PCP - CEAA users cannot see Rejected comments on PCP comment display page
* EPIC-1024 Joint PCP CEAA role CSV download. Need to list rejected documents.
* EPIC-1026 Joint PCP - comment details dialog for CEEA users

### CODE REVIEW

File: modules/project-comments/client/directives/comment-directives.js [hunk 1]
```diff
@@ -292,19 +327,27 @@ angular.module ('comment')
 					resolve: {
 						docs: function() {
 							return CommentModel.getDocuments(comment._id);
+						},
+						docs2: function() {
+							return CommentModel.getDocuments2(comment._id);
 						}
 					},
-					controller: function ($scope, $modalInstance, docs) {
+					controller: function ($scope, $modalInstance, docs, docs2) {
```
Comment (marklise) [inline modules/project-comments/client/directives/comment-directives.js]: Variable naming should be more descriptive.

File: modules/project-comments/client/directives/comment-directives.js [hunk 2]
```diff
@@ -632,48 +579,337 @@ angular.module ('comment')
...
+					var docCount = s.fileList.length;
+
+					if (!comment.author) {
+						comment.isAnonymous = true; // if the author is empty make it anonymous
+					}
+
+					if (docCount === 0 ) {
+						// We don't need to do anything but add the comment.
+						// console.log("s.comment:", s.comment);
+							CommentModel.add (s.comment)
```
Comment (marklise) [inline modules/project-comments/client/directives/comment-directives.js]: Indentation looks off.

File: modules/project-comments/client/directives/comment-directives.js [hunk 3]
```diff
@@ -632,48 +579,337 @@ angular.module ('comment')
...
+										.catch (function (err) {
+											s.step = 4;
+											$scope.$apply ();
+										});
+									}
+								});
+							}, function (response) {
+								if (response.status > 0) {
+									// docUpload.errorMsg = response.status + ': ' + response.data;
+									console.log("error data:",response.data);
```
Comment (marklise) [inline modules/project-comments/client/directives/comment-directives.js]: Remove console.logs.

File: modules/project-comments/client/directives/comment-directives.js [hunk 4]
```diff
@@ -632,48 +579,337 @@ angular.module ('comment')
...
+						.then(function() {
+							return uploadFiles(url, ctrl.package2.validFiles, comment.documents2);
+						})
+						.then(function() {
+							console.log("save comment");
+							return CommentModel.add(comment);
+						})
+						.then (function (comment) {
+							ctrl.isBusy = false;
+							console.log("go to next stage");
```
Comment (marklise) [inline modules/project-comments/client/directives/comment-directives.js]: Remove console.log

**Comment by asanchezr** [comment]
@marklise - I have addressed all your comments. Please review.

**Pull request merged**